awswrangler/s3/_write_dataset.py

```diff
@@ -268,6 +272,47 @@ def _to_partitions_distributed(  # pylint: disable=unused-argument
     boto3_session: boto3.Session,
     **func_kwargs: Any,
 ) -> Tuple[List[str], Dict[str, List[str]]]:
+    paths: List[str]
+    partitions_values: Dict[str, List[str]]
+
+    if not bucketing_info:
+        # If only partitioning (without bucketing), avoid expensive modin groupby
+        # by partitioning and writing each block as an ordinary Pandas DataFrame
+        _to_partitions_func = _to_partitions.dispatch(PandasDataFrame)
+
+        @ray_remote
+        def write_partitions(df: pd.DataFrame) -> Tuple[List[str], Dict[str, List[str]]]:
+            paths, partitions_values = _to_partitions_func(
+                # Passing a copy of the data frame because data in ray object store is immutable
+                # and that leads to "ValueError: buffer source array is read-only" during df.groupby()
+                df.copy(),
```

Comment:
Wondering how expensive this is memory wise...

Reply:
Absolutely valid concern. This depends on the size of the block but hopefully since it's only a local copy and no data ends up transferred between the nodes it would be fast.

There is no alternative fix (e.g. release the mutex or something of that kind) as far as I can tell but I'm happy to explore if you have any suggestions.

Reply:
I guess [customized serialization](https://docs.ray.io/en/latest/ray-core/objects/serialization.html#customized-serialization) could be one of the alternatives

Reply:
Good enough for now, we can explore serialisation later on